src/model.cpp

```diff
@@ -5,21 +5,31 @@
 #include "special_chars.h"
 #include <array>
 #include <string>
+#include <vector>
+#include <algorithm>
 
 void Model::defineCell(const double& grid_step, const int& max_depth){
   cell = Space(atoms, grid_step, max_depth);
   return;
 }
 
-void Model::storeAtomsInTree(){
-  atomtree = AtomTree(atoms);
+void Model::setAtomListForCalculation(std::vector<std::string> included_elements){

```

Comment:
You should pass the argument as constant reference, so the function doesn't make a copy of the vector:

```c++
void Model::setAtomListForCalculation(const std::vector<std::string>& included_elements){
```

You may need to add the `const` keyword at the end of the function for this to work.

Reply:
> You may need to add the `const` keyword at the end of the function for this to work.

Making the function const is only necessary if it will be called by const object (from what I understood) but it doesn't harm to do it so I will.

Reply:
If that works that's fine.